2-Authorization-I/1-call-graph/SPA/src/hooks/useTokenAcquisition.js

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react';
+import { InteractionRequiredAuthError } from '@azure/msal-browser';
+import { useMsal } from '@azure/msal-react';
+
+/**
+ * A custom hook for acquiring tokens using MSAL
+ * @param {Array} scopes
+ * @returns response object which contains an access token
+ */
+const useTokenAcquisition = (scopes) => {
+
+    /**
+     * useMsal is a hook that returns the PublicClientApplication instance,
+     * an array of all accounts currently signed in and an inProgress value
+     * that tells you what msal is currently doing. For more, visit:
+     * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md
+     */
+
+    const { instance, inProgress } = useMsal();
+    const account = instance.getActiveAccount();
+    const [response, setResponse] = useState(null);
+
+    useEffect(() => {
+        const getToken = async () => {
+            let token;
+            if (account && inProgress === 'none' && !response) {
+                try {
+                    token = await instance.acquireTokenSilent({
+                        scopes: scopes,
+                        account: account,
+                    });
+
+                    setResponse(token);
+                } catch (error) {
+                    if (error instanceof InteractionRequiredAuthError) {
+                        try {
+                            token = await instance.acquireTokenPopup({
```

Comment:
let's make the interaction method configurable i.e. consumers of this hook should pass in the interaction method (popup or redirect) that will be used if acquireTokenSilent fails

Reply:
Added interaction type to hook